### PR TITLE
Fix warnings in golangci linter

### PR DESCRIPTION
**Pull request by joeyyy09**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR updates the `.golangci.yaml` file to address the following warnings:
- Deprecated configuration options `linters.staticcheck.go` and `linters.stylecheck.go` are replaced with global `run.go`.
- Moved exclude files configuration from the deprecated `run.skip-files` to `issues.exclude-files`.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -27,8 +27,6 @@ linters:
   - nolintlint
   - prealloc
   - revive
-  - staticcheck
-  - stylecheck
```
Comment (wzshiming) [inline .golangci.yaml:31]: Why turn off the check, it works fine.

File: .github/workflows/golangci-lint.yaml [hunk 2]
```diff
@@ -17,4 +17,4 @@ jobs:
       uses: golangci/golangci-lint-action@v3
       with:
         version: v1.55.2
```
Comment (wzshiming) [inline .github/workflows/golangci-lint.yaml:19]: All checks are version dependent, the current version of lint has no warnings in CI, you should upgrade this version and fix the warnings.

Comment (joeyyy09) [reply .github/workflows/golangci-lint.yaml:19]: > All checks are version dependent, the current version of lint has no warnings in CI, you should upgrade this version and fix the warnings.

Okay, will look into this

File: .golangci.yaml (hunk 1 above)
Comment (joeyyy09) [reply .golangci.yaml:31]: i didnt turn off the check, static check and style check are deprecated and it's covered under the global run.go in the later versions, PFA the image where the warnings are shown
image

File: .github/workflows/golangci-lint.yaml (hunk 2 above)
Comment (joeyyy09) [reply .github/workflows/golangci-lint.yaml]: I've bumped up the version to 1.58 as per your suggestion in the new commit

Comment (wzshiming) [inline .github/workflows/golangci-lint.yaml:19]:
```suggestion
        version: v1.58.2
```

And also bump following

https://github.com/kubernetes-sigs/kwok/blob/main/hack/update-go-lint.sh
https://github.com/kubernetes-sigs/kwok/blob/main/hack/verify-go-lint.sh

File: .github/workflows/golangci-lint.yaml [hunk 3]
```diff
@@ -17,4 +17,4 @@ jobs:
...
-        args: --timeout=30m --config=.golangci.yaml
+        args: --timeout=60m --config=.golangci.yaml
```
Comment (wzshiming) [inline .github/workflows/golangci-lint.yaml:20]:
```suggestion
        args: --timeout=30m --config=.golangci.yaml
```

Comment (wzshiming) [reply .github/workflows/golangci-lint.yaml:20]: Modify `.golangci.yaml` to allow this

```log
pkg/kwok/server/debugging_logs.go:297:7: max-control-nesting: control flow nesting exceeds 5 (revive)
                                                if os.IsNotExist(err) {
                                                        continue
                                                }
```

**Comment by joeyyy09** [comment]
@wzshiming  I've fixed all the warnings and errors. Can you please have a look at them?

**Comment by wzshiming** [comment]
Still, pull-kwok-verify-main — Job failed.
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kwok/1085/pull-kwok-verify-main/1792890641768779776

File: .golangci.yaml [hunk 4]
```diff
@@ -59,9 +59,11 @@ linters-settings:
     - pkg: k8s.io/apimachinery/pkg/api/errors
       alias: apierrors
   staticcheck:
-    go: "1.22"
+   run:
```
Comment (wzshiming) [inline .golangci.yaml:62]:
```suggestion
     run:
```

File: .golangci.yaml [hunk 5]
```diff
@@ -180,17 +182,19 @@ linters-settings:
...
+  - ".*\\.zz_generated\\..*"
 run:
   concurrency: 1
   timeout: 10m
   allow-parallel-runners: false
   skip-dirs-use-default: true
-  skip-files:
-  - ".*\\.zz_generated\\..*"
-  go: "1.22"
+  go: "1.22"
```
Comment (wzshiming) [inline .golangci.yaml:200]:
```suggestion
  go: "1.22"

```

**Review by wzshiming — COMMENTED** [review]
Squash commits and rebase to main

**Pull request closed without merging**